### DISCUSSION TITLE
「分享」使用 iOS 快捷指令录入笔记，支持多图上传，支持标签选择

### DISCUSSION THREAD

**Discussion by monlor** [usememos/memos#52] [Ideas]
### 2022-07-12 适配0.2.0，支持标签选择接口
https://www.icloud.com/shortcuts/d1adc0feefe34ac28612d3ec00253072

### 2022-06-22 适配0.1.2，支持多图上传，适配新的markdown图片格式#81
https://www.icloud.com/shortcuts/6762028ebfa544cca3cfe148fe3ec848

### 2022-06-20 HEIC图片无法显示，支持上传图片时自动转换图片为JPEG
https://www.icloud.com/shortcuts/80008db4502c4b64bafba85325dad51d

### 2022-05-21 支持文本+图片+标签
作者新添加的开放接口 #55
https://www.icloud.com/shortcuts/751af4392d774caeb0fd751bc78240e0

### 仅文本+标签
https://www.icloud.com/shortcuts/abfe498129574dc49a8e2b74d6b272ec

**Comment by rebron1900** [comment]
可惜安卓目前没找到类似的工具

**Reply by yyysuo** [reply]
ios14显示添加成功，实际上快捷指令里面并没有，有解决方案吗？

**Reply by rebron1900** [reply]
不好意思，我用的安卓

**Reply by eallion** [reply]
试试 https://github.com/Waboodoo/HTTP-Shortcuts

**Comment by yyysuo** [comment]
> https://www.icloud.com/shortcuts/abfe498129574dc49a8e2b74d6b272ec

<img width="256" src="https://user-images.githubusercontent.com/10081071/169210324-b336a439-edb0-4616-8f6b-3278445e351b.jpg"/>


把加标签的询问的部分去掉了，默认直接把ios剪贴板的内容打上手机的标签生成新的memo。ios15的快捷指令无法直接导入到IOS14中，我用越狱插件pastcuts强行导入了，好像在ios14上运行起来还是有点问题，只有把加标签的询问部分删除才能正确执行。

好像现在icloud目前有点问题

```
Unknown Host
Description: Unable to locate the server for this request -- the server does not have a DNS entry. Perhaps there is a misspelling in the server name, or the server no longer exists. Double-check the name and try again.
```

**Reply by lhuanyun** [reply]
找到一个插件，另外我自己写了个ios13 14的捷径。在tg里面发过。tag自己输入，可以上传资源… 有问题自己研究。感谢

**Comment by yyysuo** [comment]
> ### 2022-05-21 支持文本+图片+标签
> 作者新添加的开放接口 #55 https://www.icloud.com/shortcuts/751af4392d774caeb0fd751bc78240e0
> 
> ### 仅文本+标签
> https://www.icloud.com/shortcuts/abfe498129574dc49a8e2b74d6b272ec

求大神做一个调用系统相机拍照并打上默认标签的快捷指令。另外ios15上面的快捷指令默认是无法导入ios14的，越狱的iphone还有解决办法，但是运行起来只能输入标签，无法输入笔记内容，有解决的办法么？

**Comment by boojack** [comment]
刚试了下，很赞！

---

关于图片这里，需要更新下格式，现在是用 markdown 的图片块语法：`![](prev_img_url)` https://github.com/usememos/memos/issues/81#issuecomment-1161145506
<img width="300" alt="image" src="https://user-images.githubusercontent.com/24653555/174953829-cb159b8f-5b28-4859-81da-003b84dc4a89.png">

**Reply by monlor** [reply]
好的，已支持，啥时候发正式版

**Reply by beyondtop** [reply]
0.2版本失效了，要更新了。

**Reply by boojack** [reply]
应该是涉及到的接口更新了，FYI：
* 新增 memo：`POST /api/memo?openId=xxx`
* 新增图片：`POST /api/resource?openId=xxx`
* 获取 tags: `GET /api/tag?openId=xxx`

**Reply by beyondtop** [reply]
谢谢，已更改，可以使用了。

**Reply by monlor** [reply]
已适配0.2.0

**Reply by beyondtop** [reply]
> 已适配0.2.0

你好，可以增加“公开”或“私人”的状态标签选择吗？

**Reply by monlor** [reply]
啥意思？你添加过这2个标签就会有选项了

**Reply by beyondtop** [reply]
> 啥意思？你添加过这2个标签就会有选项了

你好，0.2版本不是增加了PUBLIC状态选择吗？ 默认是PRIVATE。选择public，就可以不用登陆也能公开浏览。这样在快捷指令里增加这一选项，就能选择是否公开浏览，不用再回到网页上去选择。

**Reply by monlor** [reply]
我咋没看到有这个功能呢？接口里面有这个选项么 @boojack

**Reply by boojack** [reply]
接口里面暂时没有，下个版本会加上

**Reply by boojack** [reply]
已更新，在 POST /api/memo 的请求体里加了个可选 `visibility` 的字段，值为 `PRIVATE` 和 `PUBLIC`，如果不填的话默认就会是 `PRIVATE`
https://github.com/usememos/memos/blob/main/server/memo.go#L27

**Reply by beyondtop** [reply]
速度真快。谢谢

**Reply by monlor** [reply]
加这个感觉快捷指令的选项有点多了，就不快捷了，考虑加一个开关控制

**Reply by boojack** [reply]
最近有打算做“个人偏好设置”，因为需要改动 schema，所以会在下一个大版本 0.3.0 更新

**Reply by beyondtop** [reply]
> 加这个感觉快捷指令的选项有点多了，就不快捷了，考虑加一个开关控制

的确是，加个开关就方便多了。

**Reply by rebron1900** [reply]
> 应该是涉及到的接口更新了，FYI：
> 
> * 新增 memo：`POST /api/memo?openId=xxx`
> * 新增图片：`POST /api/resource?openId=xxx`
> * 获取 tags: `GET /api/tag?openId=xxx`

现在只要有openid就能使用这些接口么？比如我在前端想把数据渲染到我博客页面上，这样可能就会暴露我的openid，这种情况别人拿我的openid是不是就可以往我的memos里发送数据？

**Reply by boojack** [reply]
可以通过 GET `/api/memo?creatorId=xxx` 来获取到你的 public memos，例如 https://memos.onrender.com/api/memo?creatorId=101

**Reply by rebron1900** [reply]
是不是改吧手册什么的写一写了。

**Comment by iwyang** [comment]
安卓可以参考：https://blog.leonus.cn/2022/talkInAndroid.html
不过只能发布文字。

**Comment by lmm214** [comment]
近期是否有更新匹配 0.5 接口的计划？发图的接口变了

**Comment by jhx520** [comment]
> ### 2022-07-12 适配0.2.0，支持标签选择接口
> https://www.icloud.com/shortcuts/d1adc0feefe34ac28612d3ec00253072
> 
> ### 2022-06-22 适配0.1.2，支持多图上传，适配新的markdown图片格式#81
> https://www.icloud.com/shortcuts/6762028ebfa544cca3cfe148fe3ec848
> 
> ### 2022-06-20 HEIC图片无法显示，支持上传图片时自动转换图片为JPEG
> https://www.icloud.com/shortcuts/80008db4502c4b64bafba85325dad51d
> 
> ### 2022-05-21 支持文本+图片+标签
> 作者新添加的开放接口 #55 https://www.icloud.com/shortcuts/751af4392d774caeb0fd751bc78240e0
> 
> ### 仅文本+标签
> https://www.icloud.com/shortcuts/abfe498129574dc49a8e2b74d6b272ec

无法导入IOS14，是否适配一下14，目前应该都是15才能导入

**Comment by KeithW0726** [comment]
报错，我的是ios16，返回信息找不到词典keydata的值

**Comment by lhuanyun** [comment]
使用了一下，选择标签的功能不错哟。我ios13 根据上面同学说的插件，强制导入了ios15脚本。没有使用上的问题，完美兼容。
由于最近图片接口改了，楼主还没有更新接口，文章插入图片会有问题，还有上传文件也可能存在问题，就不尝试了。
我自己用的上次半成品发TG里，修改一下就可以正常使用了。

逻辑是:
1.（选择输入文字 选择剪切板文字 分享输入）
2.（公开 隐私 保护）
3.（直接发布 文件附件 图片附件 共享附件）

![image](https://user-images.githubusercontent.com/20707383/204852743-8b36d8fb-e228-4cb3-87ef-ed3d74be0f1e.jpeg)

![image](https://user-images.githubusercontent.com/20707383/204852800-f84abdd7-bf28-4541-b2aa-33e61c0c7dbe.jpeg)

![image](https://user-images.githubusercontent.com/20707383/204852846-e2eb970e-9f56-414b-a073-2003ada1a34a.jpeg)

标签省去了，直接输入就行。选项太多，太麻烦… 喜欢速入
